chart/kubeapps/templates/ingress.yaml

```diff
@@ -9,6 +9,11 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
   annotations:
+    {{- $rewrites := include "kubeapps.ingressRewrites" . -}}
+    {{- if ne $rewrites ""}}
+    nginx.ingress.kubernetes.io/configuration-snippet: |
+      {{ template "kubeapps.ingressRewrites" $ }}
```

Comment:
I had success with `rewrite ^([^.]*[^/])$ $1/ permanent;` as the snippet here, removing the need to cycle the paths.

Reply:
See https://stackoverflow.com/questions/645853/add-slash-to-the-end-of-every-url-need-rewrite-rule-for-nginx, this regex "[add a slash to the end] if the URI doesn't have a period and does not end with a slash"

Reply:
It seems that this rewrite has consequences since it also rewrites async calls like namespaces or releases, and tiller proxy does not seem to be able to handle that call.

![selection_802](https://user-images.githubusercontent.com/24523/50608666-24f3b700-0e82-11e9-884b-1892d52babd2.png)